pkg/issuer/acme/http/pod.go

```diff
@@ -52,13 +52,12 @@ func (s *Solver) ensurePod(ch *v1alpha1.Challenge) (*corev1.Pod, error) {
 		return existingPods[0], nil
 	}
 	if len(existingPods) > 1 {
-		errMsg := fmt.Sprintf("multiple challenge solver pods found for certificate '%s/%s'. Cleaning up existing pods.", ch.Namespace, ch.Name)
-		glog.Infof(errMsg)
+		glog.Infof("multiple challenge solver pods found for certificate '%s/%s'. Cleaning up existing pods.", ch.Namespace, ch.Name)
 		err := s.cleanupPods(ch)
 		if err != nil {
 			return nil, err
 		}
-		return nil, fmt.Errorf(errMsg)
+		return s.createPod(ch)
```

Comment:
We can't just change this to call createPod without also making the pod's name determinate else there's a risk we could enter a loop of:

* create a pod/present successfully
* resync occurs and we've not observed the new pod in the lister
* a new pod will be created - because we use GenerateName, the call to the apiservers Create endpoint will succeed, next time we loop around here, we'll delete both pods because multiple exist, and the loop re-runs.

We can avoid this by not using GenerateName in buildPod, and instead using Name (where 'name' is predictable). That way, we'll get an AlreadyExists error if we hit this race, causing the controller to resync after a back-off (which allows for the cache to become consistent)